connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -128,6 +133,36 @@ private[sql] class AvroDeserializer(
       case (INT, IntegerType) => (updater, ordinal, value) =>
         updater.setInt(ordinal, value.asInstanceOf[Int])
 
+      case (LONG, dt: TimestampType)
```

Comment:
Can we group the cases with some rules? e.g.
```
case (INT, ...)
case (INT, ...)
...
case (LONG, ...)
case (LONG, ...)
...
```

Reply:
otherwise it's very hard to review the logic if these random patter match cases.

Reply:
Shall we just match `case(Long, DatetimeType)`?

Reply:
Is `DatetimeType` the same as `TimestampType`? Why using `DatetimeType` is better?

Reply:
TimestampType/TimestampNTZType/DateType are all of DatetimeType. We can reduce duplicated code

Reply:
oh, that's a good idea!